pkg/cloud/services/network/subnets.go

```diff
@@ -158,12 +160,20 @@ func (s *Service) reconcileSubnets() error {
 			existingSubnet.DeepCopyInto(sub)
```

Comment:
This line currently breaks the name preservation, `existing` never populates `resourceID` so `resourceID` gets wiped and the `id` gets set to the subnet-id.

Propose to update the fetching of existing to only populate `resourceID` (effect for existing is that `id` and `subnetID` will become the same value), and then copy the `id` from `sub` before we do the deepcopy.

This way the `id` is preserved if it exists today, but no longer has the effect of being required to match the subnet name.

Reply:
`existing` should now populate the `resourceID ` properly